cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/artifact/ApplicationClassDeserializer.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.artifact;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+import io.cdap.cdap.api.artifact.ApplicationClass;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.Requirements;
+
+import java.lang.reflect.Type;
+
+/**
+ * JSON deserializer for {@link ApplicationClass}
+ */
+public class ApplicationClassDeserializer implements JsonDeserializer<ApplicationClass> {
+
+  private static final String REQUIREMENTS = "requirements";
+  private static final String CLASS_NAME = "className";
+  private static final String DESCRIPTION = "description";
+  private static final String CONFIG_SCHEMA = "configSchema";
+
+  @Override
+  public ApplicationClass deserialize(JsonElement json, Type typeOfT, JsonDeserializationContext context)
+    throws JsonParseException {
+    JsonObject jsonObject = json.getAsJsonObject();
+    Schema configSchema = getConfigSchema(context, jsonObject);
+    Requirements requirements = getRequirements(jsonObject.get(REQUIREMENTS), context);
+    return new ApplicationClass(getStringValue(jsonObject.get(CLASS_NAME)), getStringValue(jsonObject.get(DESCRIPTION)),
+                                configSchema, requirements);
+  }
+
+  private String getStringValue(JsonElement jsonElement) {
+    if (jsonElement == null) {
+      return null;
+    }
+    return jsonElement.getAsString();
+  }
+
+  private Schema getConfigSchema(JsonDeserializationContext context, JsonObject jsonObject) {
+    JsonElement jsonElement = jsonObject.get(CONFIG_SCHEMA);
+    if (jsonElement == null) {
+      return null;
+    }
+    return context.deserialize(jsonElement, Schema.class);
+  }
+
+  private Requirements getRequirements(JsonElement requirementJSON, JsonDeserializationContext context) {
```

Comment:
Please be consistent with the ordering of parameters between methods.

Reply:
Also, try to be consistent for the method (e.g. it takes `JsonObject` and get the field in the `getConfigSchema` method, while in here, it is expected the caller to get the field and pass it in).

Reply:
Done